### PR TITLE
  Declare peer dependency on @ember/string for compatibility with Ember >= 5

### PR DESCRIPTION
**Pull request by jelhan**
Declaring peer dependency on @ember/string to make the addon compatible with Ember >= 5.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -94,5 +94,8 @@
     "after": [
       "ember-changeset"
     ]
+  },
+  "peerDependencies": {
+    "@ember/string": "^4.0.0"
```
Comment (SergeAstapov) [inline package.json:99]: @jelhan can we keep this
```suggestion
    "@ember/string": "^3.0.0 || ^4.0.0"
```
or
```suggestion
    "@ember/string": ">=3.0.0"
```

to simplify community migration? e.g. ember-data v4.12 depends on `@ember/string` v3

File: package.json [hunk 2]
```diff
@@ -83,8 +84,7 @@
     "node": "12.* || 14.* || >= 16"
   },
   "volta": {
-    "node": "16.20.2",
-    "npm": "6.14.18"
+    "node": "20.18.0"
```
Comment (SergeAstapov) [inline package.json:87]: I guess, this needs to be in separate PR.

@jelhan mind if we keep `volta` for now as is? I intentionally kept it without changes until v2 conversion is completed

Comment (jelhan) [reply package.json:87]: The issue was that the existing lock file was not compatible with the specified NPM version. It was in v2 format not supported by NPM v6. Pinning to a recent version and regenerating lock file was the easiest solution.

Comment (jelhan) [reply package.json]: CI was failing. Likely an incompatibility caused by very outdated node 12 used in CI. I restored the original pinned versions of Node and NPM and regenerated the lockfile with that version.

**Comment by jelhan** [comment]
Do you consider adding an additional peer dependency as a breaking change? If so, we may want to do other breaking changes in the same release.

Potential breaking changes which we may want to ship are

* Drop support for node < 18
* Drop support for Ember < 4.12
* Drop support for ember-get-config < 2

**Comment by SergeAstapov** [comment]
@jelhan as this addon already depends on `ember-auto-import`, the conversion to v2 won't be a breaking change itself.
and as a result, v2 addons do not need to declare supported node.js versions, hence updagin Node.js in CI would be non-breaking change for that reason.

Was thinking about dropping support for older Ember versiosn to simplify CI in case if I can't get CI work fine with those old versions.

Considering this change, sounds like we should plan new major as a result of many updates discussed here.

**Comment by SergeAstapov** [comment]
@jelhan as for CI failures, that's a problem for v5 and up, I haven't get to fixing it in this repo as fixed it in only in `ember-changeset`.

@jelhan if you'd like to help, I planned to have a separate PR just to bring CI to green state without any major/breaking changes

**Comment by jelhan** [comment]
> @jelhan as for CI failures, that's a problem for v5 and up, I haven't get to fixing it in this repo as fixed it in only in `ember-changeset`.

Got it.

> @jelhan if you'd like to help, I planned to have a separate PR just to bring CI to green state without any major/breaking changes

I have only limited open source time available. If you are already at it, that's great. We can park this PR until CI is fixed.
